ComSemApp/commenting_algorithm.py

```diff
@@ -0,0 +1,199 @@
+    for word in correct_expression_list:
+        if word not in present_words:
+            present_words.append(word)
+            number_present.append(1)
+        else:
+            temp = present_words.index(word)
+            number_present[temp] += 1
+
+    # Determines if there are too many of a word
+    for word in expression_list:
+        if word in present_words and word not in extra_present_words:
+            temp = present_words.index(word)
+            if(number_present[temp] == 0):
+                extra_present_words.append(word)
+            else:
+                number_present[temp] -= 1
+
+    # Determines if there is too little of a word
+    for word in expression_list:
+        if word in present_words and word not in missing_present_words:
+            temp = present_words.index(word)
+            if(number_present[temp] != 0):
+                missing_present_words.append(word)
+
+
+    return extra_present_words, missing_present_words
+
+
+def wrong_position(expression, correct_expression):
+    """
+    A function that returns a list, of strings, of words that are in expression, but not in the same positon as 
+    correct_expression.
+    Args:
+        expression (string): The student provided version of the expression, the reformulation.
+        correct_expression (string): The correct expression the given expression will be compared to.
+    
+    Returns:
+        words_in_wrong_position (list): A list of strings that stores words in the wrong position in expression
+    """
+    lower_expression = expression.lower()
+    expression_list = lower_expression.split()
+    lower_correct_expression = correct_expression.lower()
+    correct_expression_list = lower_correct_expression.split()
+    words_in_wrong_position = []
+    correct_expression_length = len(correct_expression_list)
+    expression_length = len(expression_list)
+
+    if(expression_length > correct_expression_length):
+        min_length = correct_expression_length
+    else:
+        min_length = expression_length
+
+    for x in range(min_length):
+        if correct_expression_list[x] != expression_list[x]:
+            words_in_wrong_position.append(correct_expression_list[x])
+
+    return words_in_wrong_position
+            
+
+
+def get_comments(expression, correct_expression):
+    """
+    This function utlizes the other functions in this file to find errors and provide feedback on those errors for students to 
+    improve. This funciton is used in the review page to get the higlighsts and comments on student work.
+    Args:
+        expression (string): The student provided version of the expression, the reformulation.
+        correct_expression (string): The correct expression the given expression will be compared to.
+    
+    Returns:
+        comments (list): A list of dictionaries the indexs of errors made in the sentence and comments to identify them.
+    """
+    comments = []
+    lower_expression = expression.lower()
```

Comment:
once you make the lower case versions, just pass those in everywhere. Do this for both correct and reformulation